Sources/Lifecycle/Lifecycle.swift

```diff
@@ -311,11 +324,17 @@ public class ComponentLifecycle: LifecycleTask {
 
         self.stateLock.lock()
         switch self.state {
-        case .idle:
+        case .idle(let tasks) where tasks.isEmpty:
             self.state = .shutdown(nil)
             self.stateLock.unlock()
             defer { self.shutdownGroup.leave() }
             callback(nil)
+        case .idle(let tasks):
+            self.stateLock.unlock()
+            let queue = DispatchQueue.global() // is this the best we can do?
```

Comment:
As of now the queue that is used in the other states is supplied during `start()`. Maybe we could supply this queue in the initializer and set it as a `let`? This way we could access the same queue for stop in any situation?

Reply:
remove this shutdown semantics